### ISSUE TITLE
Minifier is removing unused function args in production run

### ISSUE THREAD

**Issue by jmdsg** [meteor/meteor#11619]
- Version: I am using meteor 2.3.5, but also tried on meteor 2.2.1 and it behaves the same.
- Last version it worked: Not really sure.
- The operating system: Windows 10 Version 10.0.19043 Build 19043
- The expected behavior: Unused arguments not being deleted in the minified code, since some libraries i use (ramda) rely on `fn.length`.
- The actual behavior: Unused arguments are being deleted in the minified code.
- Steps to reproduce:

```bash
meteor create --release 2.3.5 meteor-2.3.5
cd meteor-2.3.5

meteor npm i color ramda

echo "import * as R from 'ramda';" >> client/main.jsx
echo "import Color from 'color';" >> client/main.jsx
echo "const fn = R.curry((a, b) => a + b);" >> client/main.jsx
echo "console.debug(Color);" >> client/main.jsx
echo "console.debug(R.flip(fn)(1)(2));" >> client/main.jsx

meteor run --production
```

The problem seems to be that after an update of the color package from `3.2.1` to the version `4.0.1` it is now causing some kind of problem with the minifier, and is causing the unused arguments to be removed in some ramda functions.

In the reproduction steps, if either the color package is downgraded to `3.2.1`, or removed, then everything works ok. The main difference that i see between those versions is the use of `for...of`, that im not exactly sure is supported (even though, i used it locally and it worked).

This is a part of the expected minified ramda code (formatted here just to show):

```javascript
case 10:
    return function(e, n, r, l, i, a, u, s, o, c) {
        return t.apply(this, arguments)
    };
default:
    throw new Error("First argument to _arity must be a non-negative integer no greater than ten")
```

But, this is the present code.

```javascript
case 10:
    return function() {
        return t.apply(this, arguments)
    };
default:
    throw new Error("First argument to _arity must be a non-negative integer no greater than ten")
```

**Comment by StorytellerCZ** [comment]
@jmdsg can you please give it a try in the latest 2.4 RC, minifiers were updated for that release.

**Comment by StorytellerCZ** [comment]
@jmdsg can you try this PR if it fixes the issue: https://github.com/meteor/meteor/pull/11630

**Comment by StorytellerCZ** [comment]
From terser docs:
```
keep_fargs (default: true) -- Prevents the compressor from discarding unused function arguments. You need this for code which relies on Function.length.
```
So it should keep the unused args.

**Comment by jmdsg** [comment]
> From terser docs:
> 
> ```
> keep_fargs (default: true) -- Prevents the compressor from discarding unused function arguments. You need this for code which relies on Function.length.
> ```
> 
> So it should keep the unused args.

I read about that as well, but afaik, keep_fargs is enabled, and it is actually woking as expected, except with this particular case using the color package version 4.0.1, and potentially with some similar cases that we don't know about. So that makes me think, that something in the code of color@4.0.1, makes the terser to have an unexpected behavior, but i haven't been able to isolate exactly which part.

> @jmdsg can you try this PR if it fixes the issue: #11630

Yes absolutely, i will try it today if i can, or tomorrow.

**Comment by filipenevola** [comment]
Hi, `minifier-js@2.7.1` and `standard-minifier-js@2.7.1` are published.

A new `meteor update --release 2.4` should update these packages for you if you are already running Meteor 2.4.
